genestack_client/files_util.py

```diff
@@ -120,16 +121,41 @@ def collect_initializable_files_in_container(self, accession):
         """
         return self.invoke('collectInitializableFilesInContainer', accession)
 
-    def get_file_children(self, container_accession):
+    def count_file_children(self, container_accession):
+        """
+        Count children of a container (not recursive).
+        :param container_accession: accession of container
+        :type container_accession: str
+        :return: number of children
+        """
+        return self.invoke('countFileChildren', container_accession)
+
+    def get_file_children(self, container_accession, show_progress=False):
         """
         Return accessions of files linked to current container.
 
         :param container_accession:  accession of container
         :type container_accession: str
+        :param show_progress: should a progress indicator be shown
+        :type show_progress: bool
         :return: list of accessions
         :rtype: list
         """
-        return self.invoke('getFileChildren', container_accession)
+        all_files = []
+        count = 0
+        while True:
+            if show_progress:
+                sys.stdout.write('\rRetrieving container children (%d)...' % count)
```

Comment:
Depends who you ask. In sublime text it looks good because ^R is rendered as a newline character, but in `less`, I see that:
```
^MRetrieving container children (0)...^MRetrieving container children (500)...^M
```